blark/iec.lark

```diff
@@ -323,6 +326,7 @@ array_spec_init: [ indirection_type ] array_specification [ ":=" array_initializ
 array_specification: "ARRAY"i "[" subrange ( "," subrange )* "]" "OF"i _array_spec_type_name
 
 _array_spec_type_name: STRING_TYPE
+                     | function_call
```

Comment:
Referring back to your example -
```
        currentChannel : ARRAY[APhase..CPhase] OF class_baseVector(SIZEOF(vector_t),0);
```
This is another new one to me. Interesting...

Reply:
Agreed.... One of my colleagues had pointed it out to me in an older sample of some logic we use. Was quite the "head-scratcher" for me! hah!